frontend/src/components/NewCourseForm.tsx

```diff
@@ -36,10 +37,10 @@ export default function NewCourseForm(props:Readonly<NewCourseFormProps>){
         props.closeDialog();
     }
 
-    function handleChangeCourseType(event:SelectChangeEvent){
+    async function handleChangeCourseType(event:SelectChangeEvent){
```

Comment:
There don't seem to be any asynchronous operations in this function, so you can likely remove the async keyword.